vertx-web/src/main/java/io/vertx/ext/web/handler/impl/BodyHandlerImpl.java

```diff
@@ -97,6 +97,12 @@ private BHandler(RoutingContext context) {
       });
     }
 
+    private void makeUploadDir(FileSystem fileSystem) {
+      if (!fileSystem.existsBlocking(uploadsDir)) {
```

Comment:
If the configured directory is a file, you get the following somewhat descriptive error message already during execution, e.g.: 
Caused by: java.nio.file.FileSystemException: <vertxpath>/file-uploads/c9a8e424-a225-4c39-aecd-e9ab1c0b8d5a: Not a directory

The check for a directory was not present in the original code either. I can still add it with a more descriptive error message, but it would require an additional call to the file-system. Let me know if you still want the additional check, or the existing thrown is sufficiently expressive.